### PR TITLE
[4.0] [com_finder] Disable search statistics by default

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

In the administrator part of Smart Search, search statistics option appears to be disabled but in frontend statistics are still gathered. This changes default value of statistics option in frontend to disabled.

Also renames `logging_enabled` parameter to `gather_search_statistics` to avoid confusion with `enable_logging`.

### Testing Instructions

In backend go to Components -> Smart Search -> Statistics.
Notice the warning:

>Gathering of statistics is disabled. Enable it in the Options.

Go to frontend. Perform a search using Smart Search.
Go back to statistics page.

### Expected result

No entries logged.

### Actual result

Search entry logged.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 3b275519f3c0a8008754b65f1e27eb5d184d0592<br /><br />Tested successfully in Nightly Build of May 20.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29164">issues.joomla.org/tracker/joomla-cms/29164</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3b275519f3c0a8008754b65f1e27eb5d184d0592<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29164">issues.joomla.org/tracker/joomla-cms/29164</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29164">issues.joomla.org/tracker/joomla-cms/29164</a>.</sub>

**Comment by richard67** [comment]
@SharkyKZ ~~Could you update the branch of this PR to latest 4.0-dev of the CMS? It needs to do that in order to make drone pass, and this is needed for merging (except someone merges who has permission to force merge, which I haven't).~~ Wait, I first try it with restarting drone.

File: components/com_finder/src/Helper/FinderHelper.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ class FinderHelper
 	 */
 	public static function logSearch(Query $searchquery, $resultCount = 0)
 	{
-		$enable_log_searches = ComponentHelper::getParams('com_finder')->get('logging_enabled', 1);
+		$enable_log_searches = ComponentHelper::getParams('com_finder')->get('gather_search_statistics', 0);
```
Comment (zero-24) [inline components/com_finder/src/Helper/FinderHelper.php]: should't we update the variable names than too?

**Review by zero-24 — CHANGES_REQUESTED** [review]

File: components/com_finder/src/Helper/FinderHelper.php (hunk 1 above)
Comment (SharkyKZ) [reply components/com_finder/src/Helper/FinderHelper.php]: Makes no difference. But this one can be removed.

**Comment by zero-24** [comment]
Merging thanks @SharkyKZ

**Pull request merged**